Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIScripting.mm

```diff
@@ -88,9 +89,277 @@
+
+        @"function getBackgroundColor() { return window.getComputedStyle(document.body).getPropertyValue('background-color') }",
+        @"function getFontSize() { return window.getComputedStyle(document.body).getPropertyValue('font-size') }",
+
+        @"function logMessage() { console.log('Logging message') }",
+
+        @"await browser.scripting.insertCSS( { target: { tabId: tabId, frameIds: [ 0 ] }, files: [ 'backgroundColor.css' ] })",
+        @"results = await browser.scripting.executeScript( { target: { tabId: tabId, allFrames: true }, func: getBackgroundColor })",
+        @"browser.test.assertEq(results[0].result, pinkValue)",
+        @"browser.test.assertEq(results[1].result, blueValue)",
+
+        @"await browser.scripting.removeCSS( { target: { tabId: tabId, frameIds: [ 0 ] }, files: [ 'backgroundColor.css' ] })",
+        @"results = await browser.scripting.executeScript( { target: { tabId: tabId, allFrames: true }, func: getBackgroundColor })",
+        @"browser.test.assertEq(results[0].result, transparentValue)",
+        @"browser.test.assertEq(results[1].result, blueValue)",
+
+        @"await browser.scripting.insertCSS( { target: { tabId: tabId, frameIds: [ 0 ] }, css: 'body { background-color: pink !important }' })",
+        @"results = await browser.scripting.executeScript( { target: { tabId: tabId, allFrames: true }, func: getBackgroundColor })",
+        @"browser.test.assertEq(results[0].result, pinkValue)",
+        @"browser.test.assertEq(results[1].result, blueValue)",
+
+        @"await browser.scripting.removeCSS( { target: { tabId: tabId, frameIds: [ 0 ] }, css: 'body { background-color: pink !important }' })",
+        @"results = await browser.scripting.executeScript( { target: { tabId: tabId, allFrames: true }, func: getBackgroundColor })",
+        @"browser.test.assertEq(results[0].result, transparentValue)",
+        @"browser.test.assertEq(results[1].result, blueValue)",
+
+        // FIXME: <https://webkit.org/b/262491> Test with subframe once there's support for style injections for specific frame Ids.
+
+        @"browser.test.notifyPass()"
+    ]);
+
+    static auto *backgroundColor = @"body { background-color: pink !important }";
+    static auto *fontSize = @"body { font-size: 104px }";
+
+    auto extension = adoptNS([[_WKWebExtension alloc] _initWithManifestDictionary:scriptingManifest resources:@{
+        @"background.js": backgroundScript,
+        @"backgroundColor.css": backgroundColor,
+        @"fontSize.css": fontSize,
+    }]);
+    auto manager = adoptNS([[TestWebExtensionManager alloc] initForExtension:extension.get()]);
+
+    auto *urlRequest = server.requestWithLocalhost();
+    auto *url = urlRequest.URL;
+    auto *matchPattern = [_WKWebExtensionMatchPattern matchPatternWithScheme:url.scheme host:url.host path:@"/*"];
+    [manager.get().context setPermissionStatus:_WKWebExtensionContextPermissionStatusGrantedExplicitly forMatchPattern:matchPattern];
```

Comment:
Can just do:

```
[manager.get().context setPermissionStatus:_WKWebExtensionContextPermissionStatusGrantedExplicitly forURL:urlRequest.URL];
```

It will make the pattern for you.